src/sdk/namespace_fs.js

```diff
@@ -815,10 +824,16 @@ class NamespaceFS {
                 try {
                     dbg.warn('NamespaceFS: open dir streaming', dir_path, 'size', cached_dir.stat.size);
                     dir_handle = await nb_native().fs.opendir(fs_context, dir_path); //, { bufferSize: 128 });
+                    // fix : --recursive listing failes when paggination stop at last elemet of the the dir.
+                    if (next_pos && BigInt(next_pos) !== BigInt(2147483647)) {
```

Comment:
Why we need to call the same function twice here? what was the error that caused the failure?